go/mysql/flavor_mariadb.go

```diff
@@ -152,6 +153,36 @@ func (mariadbFlavor) status(c *Conn) (SlaveStatus, error) {
 	return parseMariadbSlaveStatus(resultMap)
 }
 
+// masterStatus is part of the Flavor interface.
+func (m mariadbFlavor) masterStatus(c *Conn) (MasterStatus, error) {
+	qr, err := c.ExecuteFetch("SHOW MASTER STATUS", 100, true /* wantfields */)
+	if err != nil {
+		return MasterStatus{}, err
+	}
+	if len(qr.Rows) == 0 {
+		// The query returned no data, meaning the server
+		// is not configured as a master.
+		return MasterStatus{}, ErrNotMaster
+	}
+
+	resultMap, err := resultToMap(qr)
+	if err != nil {
+		return MasterStatus{}, err
+	}
+
+	// SHOW MASTER STATUS does not return the executed GTIDSet for MariaDB, so we patch it in.
+	qr, err = c.ExecuteFetch("SELECT @@GLOBAL.gtid_binlog_pos", 1, false)
+	if err != nil {
+		return MasterStatus{}, err
+	}
+	if len(qr.Rows) != 1 || len(qr.Rows[0]) != 1 {
+		return MasterStatus{}, vterrors.Errorf(vtrpc.Code_INTERNAL, "unexpected result format for gtid_binlog_pos: %#v", qr)
+	}
+	resultMap["Gtid_Binlog_Pos"] = qr.Rows[0][0].ToString()
```

Comment:
I'd rather add the value only at the level of our internal struct (after calling `parseMariadbMasterStatus`) instead of tricking ourselves into thinking MariaDB returned an imaginary field.

Reply:
I debated this for a while. If we did that, then there is zero need for a `parseMariaDBMasterStatus` function. We can just call the shared `parseMasterStatus` function and then call `m.masterGTIDSet(c)` and add it on in. Thoughts?

Reply:
Ping @enisoc follow up on this?

Reply:
Ok, will adjust right now.

Reply:
Adjusted.